app/Console/Commands/AddSponsor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use \App\Sponsor;
+use \App\User;
+use Auth;
+
+class AddSponsor extends Command
+{
+    /** @var User $user */
+    private $user;
+
+    /** @var Sponsor $sponsor */
+    private $sponsor;
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'arc:addSponsor
+                                {name : the name of the sponsoring entity}
+                                {shortcode : the prefix that will be part of voucher codes}  
+                                {email : email address of User who\'s responsible for this action}
+                                ';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Adds a sponsoring entity to the available sponsors';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Find user
+        $this->user = User::where('email', $this->argument('email'))->first();
+
+        // Check if there's a Pre-exisiting sponsor.
+        $this->sponsor = Sponsor::where('name', $this->argument('name'))
+            ->orWhere('shortcode', $this->argument('shortcode'))
+            ->first();
+
+        switch (true) {
+            case (!$this->user):
+                exit("Can't find that User.\n");
+                break;
+            case ($this->sponsor):
+                exit("Sponsor " .
+                    $this->sponsor->name .
+                    ", " .
+                    $this->sponsor->shortcode .
+                    " exists, exit without change.\n");
+
+            default:
+                // Check the user is happy to proceed
+                if (!$this->warnUser()) {
+                    exit("Exit without change.\n");
+                };
+                // Log the user in.
+                Auth::login($this->user);
```

Comment:
Is it possible or desired to prompt for password - or is that taking it too far?

Reply:
Too far i think; you need console access to do this, but at some point we might want to know whom was assigned responsibility for it (when we get round to finishing the lifecycle)

Reply:
Agree